### PR TITLE
Remove non standard methods

### PR DESCRIPTION
**Pull request by eregon**
And test for method parity.

### CODE REVIEW

File: spec/truffle.mspec [hunk 1]
```diff
@@ -18,7 +22,8 @@ class MSpecScript
 
   set :target, "#{TRUFFLERUBY_DIR}/bin/truffleruby"
 
-  unless ARGV.include?('-t')  # No flags set if Ruby binary specified via -t.
+  # No flags set if a Ruby binary is specified via -t
+  if !child_process? and !ARGV.include?('-t')
```
Comment (nirvdrum) [inline spec/truffle.mspec]: I don't know if we have a Rubocop rule for this, but I thought we settled on `&&` for simple boolean expressions like this.

File: spec/truffle.mspec [hunk 2]
```diff
@@ -148,8 +153,7 @@ class MSpecScript
   set :files, get(:language) + get(:core) + get(:library) + get(:truffle)
 end
 
-is_child_process = ENV.key? "MSPEC_RUNNER"
-if i = ARGV.index('slow') and ARGV[i-1] == '--excl-tag' and is_child_process
+if i = ARGV.index('slow') and ARGV[i-1] == '--excl-tag' and MSpecScript.child_process?
```
Comment (nirvdrum) [inline spec/truffle.mspec]: I don't know if we have a Rubocop rule for this, but I thought we settled on `&&` for simple boolean expressions like this.

File: src/main/ruby/core/complex.rb [hunk 3]
```diff
@@ -303,7 +303,7 @@ def rationalize(eps = nil)
   def to_s
     result = real.to_s
 
-    if imag.kind_of?(Float) ? !imag.nan? && imag.signbit? : imag < 0
+    if imag < 0 or imag.equal?(-0.0)
```
Comment (nirvdrum) [inline src/main/ruby/core/complex.rb]: I don't know if we have a Rubocop rule for this, but I thought we settled on `||` for simple boolean expressions like this.

File: src/main/ruby/core/float.rb [hunk 4]
```diff
@@ -127,7 +120,7 @@ def to_r
   def arg
     if nan?
       self
-    elsif signbit?
+    elsif self < 0 or equal?(-0.0)
```
Comment (nirvdrum) [inline src/main/ruby/core/float.rb]: I don't know if we have a Rubocop rule for this, but I thought we settled on `||` for simple boolean expressions like this.

**Comment by nirvdrum** [comment]
I'm surprised we didn't have an `@CoreMethod`s that were missing the `@NonStandard` annotation.

**Comment by eregon** [comment]
@nirvdrum I only fixed a few core classes, it takes a fair amount of time to remove methods safely.
Also I avoided moving too much things between Fixnum/Bignum/Integer since I would just end up reverting it for 2.4 with Integer unification.

File: spec/truffle.mspec (hunk 2 above)
Comment (eregon) [inline spec/truffle.mspec]: We don't, please add one if you think it matters.
I don't want to change more existing code than needed in this PR.

**Comment by nirvdrum** [comment]
Are you planning on adding this to CI as well?

File: spec/truffle.mspec (hunk 2 above)
Comment (eregon) [inline spec/truffle.mspec]: I fixed the two cases in core I added.

**Comment by eregon** [comment]
It's a spec like any other truffle spec, so already in CI.

File: spec/tags/truffle/methods_tags.txt [hunk 5]
```diff
@@ -0,0 +1,5 @@
+slow:Public methods on BasicObject are the same as on MRI
+slow:Public methods on Object are the same as on MRI
+slow:Public methods on Enumerable are the same as on MRI
+slow:Public methods on Enumerator are the same as on MRI
+slow:Public methods on Numeric are the same as on MRI
```
Comment (chrisseaton) [inline spec/tags/truffle/methods_tags.txt]: Why are they slow?

Comment (eregon) [inline spec/tags/truffle/methods_tags.txt]: They need `ruby_exe` to ensure that no libraries are required or previous specs that could add extra methods on core classes/modules.

**Pull request merged**